### PR TITLE
Fix Style of Channels/List to group architectures in Edit Group

### LINKED ISSUE #134: Make the channel's architectures clear when assigning a channel to a group
When assigning a channel to a group, the channels list shows only the channels' names.
This was fine when there were no architectures and only around 3 or 4 channels. But it can be very confusing now if more than one architecture is available.

Here is a screenshot of the current state things can get into:
Screenshot from 2020-03-03 13-52-54

We need to have have those channels indicating their architecture clearly.
This can be done by using the secondary text in each list item or, ideally, by grouping the channels by architecture in the list.

### PR DESCRIPTION
**Pull request by ashu8912**
fixes #134

### CODE REVIEW

File: frontend/src/js/components/Groups/EditDialog.js [hunk 1]
```diff
@@ -104,12 +109,17 @@ function EditDialog(props) {
...
               </MenuItem>
-              {channels.map((channelItem) => {
-                return (<MenuItem value={channelItem.id} key={channelItem.id}>
-                          {channelItem.name}
-                        </MenuItem>);
+              {
+       channels.map((channelItem) => {
+        return (<MenuItem value={channelItem.id} key={channelItem.id}>
+                 <ListItemText primary={channelItem.name} secondary={ARCHES[channelItem.arch]}/>
+
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: Remove this blank line.

File: frontend/src/js/components/Groups/EditDialog.js [hunk 2]
```diff
@@ -104,12 +109,17 @@ function EditDialog(props) {
...
-                        </MenuItem>);
+              {
+       channels.map((channelItem) => {
+        return (<MenuItem value={channelItem.id} key={channelItem.id}>
+                 <ListItemText primary={channelItem.name} secondary={ARCHES[channelItem.arch]}/>
+                 
+                </MenuItem>)
+                       
+                   
+
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: Please remove these blank lines.

File: frontend/src/js/components/Groups/EditDialog.js [hunk 3]
```diff
@@ -104,12 +109,17 @@ function EditDialog(props) {
               <MenuItem value="" key="">
                 None yet
               </MenuItem>
-              {channels.map((channelItem) => {
-                return (<MenuItem value={channelItem.id} key={channelItem.id}>
-                          {channelItem.name}
-                        </MenuItem>);
+              {
+       channels.map((channelItem) => {
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: This new block is not aligned with the rest. Please make sure it is.

File: frontend/src/js/components/Groups/EditDialog.js [hunk 4]
```diff
@@ -104,12 +109,17 @@ function EditDialog(props) {
...
+        return (<MenuItem value={channelItem.id} key={channelItem.id}>
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: Nitpick: I know there was a return before. But if we remove the return (and the arrow function `{}`), then the component is directly returned and this makes it more readable IMO.

File: frontend/src/js/components/Groups/EditDialog.js [hunk 5]
```diff
@@ -68,7 +70,10 @@ function EditDialog(props) {
     props.onHide();
   }
 
+
+
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: Please remove these blank lines.

File: frontend/src/js/components/Groups/EditDialog.js [hunk 6]
```diff
@@ -68,7 +70,10 @@ function EditDialog(props) {
...
   function renderForm({values, status, setFieldValue, isSubmitting}) {
+
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: This also doesn't seem necessary (unless we should use it as a convention, and then we could probably leave the original style and do a global restyle once we have a style.).

**Comment by ashu8912** [comment]
Screenshot 2020-03-06 at 8 11 17 PM

**Comment by ashu8912** [comment]
@joaquimrocha I have added the desired changes, also @joaquimrocha i think its time that we setup prettier and eslint as well but before that i also want to discuss about converting the project to use CRA bootstrapped version rather than manual webpack

**Comment by joaquimrocha** [comment]
> @joaquimrocha I have added the desired changes, also @joaquimrocha i think its time that we setup prettier and eslint as well but before that i also want to discuss about converting the project to use CRA bootstrapped version rather than manual webpack

Yes. As I mentioned before, that's definitely in my plans.
I have now opened this issue:
https://github.com/kinvolk/nebraska/issues/149

File: frontend/src/js/components/Groups/EditDialog.js [hunk 7]
```diff
@@ -17,10 +17,12 @@ import React from 'react';
 import * as Yup from 'yup';
 import { applicationsStore } from '../../stores/Stores';
 import TimezonePicker, { DEFAULT_TIMEZONE } from '../Common/TimezonePicker';
+import { ListItemText } from '@material-ui/core';
+import { ARCHES } from '../../constants/helpers';
 
 function EditDialog(props) {
   const isCreation = Boolean(props.create);
-
+
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: Please make sure your editor doesn't include trailing spaces.

File: frontend/src/js/components/Groups/EditDialog.js [hunk 8]
```diff
@@ -69,8 +71,8 @@ function EditDialog(props) {
   }
 
   function renderForm({values, status, setFieldValue, isSubmitting}) {
-    const channels = props.data.channels ? props.data.channels : [];
 
+    const channels = props.data.channels ? props.data.channels : [];
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: You don't have to make this change (maybe it was your editor's formatter).

File: frontend/src/js/components/Groups/EditDialog.js [hunk 9]
```diff
@@ -104,13 +106,12 @@ function EditDialog(props) {
...
-                        </MenuItem>);
-              })
-              }
-            </Field>
+            {channels.map((channelItem) =>
+              <MenuItem value={channelItem.id} key={channelItem.id}>
+               <ListItemText primary={channelItem.name} secondary={ARCHES[channelItem.arch]}/>
+              </MenuItem>)
+            })}
+             </Field>
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/EditDialog.js]: The alignment is still different than the original. It should be at the same level of the `</MenuItem>` above.
(I know, we need to add a linter/formatter)

Comment (ashu8912) [reply frontend/src/js/components/Groups/EditDialog.js]: @joaquimrocha i have fixed the formatting please review.

**Comment by joaquimrocha** [comment]
The formatting was still off. I pushed an amended commit with a fixed version (please check it out).
Merging now.

**Pull request merged**